### DISCUSSION TITLE
64 band spectrum analyzer

### DISCUSSION THREAD

**Discussion by yliniemi** [pschatzmann/ESP32-A2DP#202] [Show and tell]
I made a fast and accurate spectrum analyzer. You can actually tell which note is playing.
I'm doing Fourier analysis with 4096 samples with 85% overlap. I get 70 fps. The accuracy comes from the Kaiser 3 windowing I do before the FFT.

This is just a prototype. I'll be adding features one by one but the algorithm is done.

Thank you for an amazing library <3
I've had a great deal of fun with it in the last two months.

https://www.youtube.com/watch?v=avjwpHCYAm0&t=105s

**Comment by pschatzmann** [comment]
Cool. Which FFT Library do you recommend?

**Reply by yliniemi** [reply]
I use Robin Scheibler's fft library modified for Arduino IDE
https://github.com/yash-sanghvi/ESP32/tree/master/FFT_on_ESP32_Arduino

It deals with real input and gives complex output. I chose it over ArduinoFFT because it's way faster and uses way less ram.

**Comment by grayoctagon** [comment]
That looks amazing. 

It would be very nice to have the .ino file to use in other Projects :)

**Reply by JoaoPedroAssis** [reply]
@yliniemi I just now found this discussion and was impressed by you project! Could you share the source code? I would be extremely useful in my own project. thanks!

**Comment by pschatzmann** [comment]
If you are looking for an easy way to do FFT I can recommend my [audio-tools library](https://github.com/pschatzmann/arduino-audio-tools/wiki/FFT)
And here is an [example with an output to an LED Matrix](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-basic-api/basic-a2dp-fft-led/basic-a2dp-fft-led.ino) with only 55 lines of code!
